src/Simplex_tree/include/gudhi/Simplex_tree.h

```diff
@@ -154,13 +157,15 @@ class Simplex_tree {
 
   struct Filtration_simplex_base_real {
     Filtration_simplex_base_real() : filt_(0) {}
+    Filtration_simplex_base_real(Filtration_value f) : filt_(f) {}
     void assign_filtration(Filtration_value f) { filt_ = f; }
     Filtration_value filtration() const { return filt_; }
    private:
     Filtration_value filt_;
   };
   struct Filtration_simplex_base_dummy {
     Filtration_simplex_base_dummy() {}
+    Filtration_simplex_base_dummy([[maybe_unused]] Filtration_value f) {}
```

Comment:
Same comment as above, shouldn't we have the same assertion as in `assign_filtration`?